SWO log output does not compile for STM32WB55

Building Zephyr with SWO log output for STM32WB55 target enabled fails due to the missing definition.

For example, hello_world sample fails with:
`west build -b nucleo_wb55rg samples/hello_world -DCONFIG_LOG=y -DCONFIG_LOG_PRINTK=y -DCONFIG_LOG_BACKEND_SWO=y`

```
/home/alex/zephyr-upstream/zephyr/soc/arm/st_stm32/common/soc_config.c: In function 'st_stm32_common_config':
/home/alex/zephyr-upstream/zephyr/soc/arm/st_stm32/common/soc_config.c:30:17: error: 'DBGMCU_CR_TRACE_MODE_Msk' undeclared (first use in this function); did you mean 'DBGMCU_CR_TRACE_IOEN_Msk'?
   30 |  DBGMCU->CR &= ~DBGMCU_CR_TRACE_MODE_Msk;
```

This happens because for some STM32 MCUs DBGMCU register does not have TRACE_MODE bits, which are responsible for Asynchronous/Synchronous trace mode selection.  For them it is enough to only set DBGMCU_CR_TRACE_IOEN bit.

The issue can be fixed with the following amendment to `soc/arm/st_stm32/common/soc_config.c`:
```
#ifdef CONFIG_LOG_BACKEND_SWO
#if defined(DBGMCU_CR_TRACE_MODE)
	/* TRACE pin assignment for asynchronous mode */
	DBGMCU->CR &= ~DBGMCU_CR_TRACE_MODE_Msk;
#endif
	/* Enable the SWO pin */
	DBGMCU->CR |= DBGMCU_CR_TRACE_IOEN;
#endif
```
Tested on nucleo_wb55rg board.

--- a/soc/arm/st_stm32/common/soc_config.c
+++ b/soc/arm/st_stm32/common/soc_config.c
@@ -26,11 +26,13 @@
 static int st_stm32_common_config(const struct device *dev)
 {
 #ifdef CONFIG_LOG_BACKEND_SWO
-	/* TRACE pin assignment for asynchronous mode */
-	DBGMCU->CR &= ~DBGMCU_CR_TRACE_MODE_Msk;
-	/* Enable the SWO pin */
-	DBGMCU->CR |= DBGMCU_CR_TRACE_IOEN;
+	/* Enable SWO trace asynchronous mode */
+#if defined(CONFIG_SOC_SERIES_STM32WBX)
+	LL_DBGMCU_EnableTraceClock();
+#else
+	LL_DBGMCU_SetTracePinAssignment(LL_DBGMCU_TRACE_ASYNCH);
 #endif
+#endif /* CONFIG_LOG_BACKEND_SWO */
 
 #if defined(CONFIG_USE_SEGGER_RTT)
 	/* On some STM32 boards, for unclear reason,

